flybywiresim/aircraft
# refactor: unify navigraph login

## Summary of Changes

* [x] Unifies Navigraph and SimBrief login into one by default
* [x] Moves Navigraph login UI to 3rd Party Options
* [x] Adds option to override SimBrief user ID to use a different account than the main Navigraph account

## Testing instructions


### **This PR cannot be tested using the normal method of downloading builds. Please check the QA channel on Discord for a build link.**

1. Test out the SimBrief import without changing anything (this step requires that you have a Navigraph account already linked in the flyPad)
   - Go to Settings -> Third Party Options -> See if your navigraph account is detected.
   - Press the import button on the dashboard. Verify that it works.
2. Test out the Navigraph account login flow
   - Go to Settings -> Third Party Options -> Unlink your existing account if applicable
   - Verify button changes to allow you to link an account.
   - Press button and follow instructions on screen. Verify you are taken back to the Third Party Options page and that your account username, subscription status and "unlink account" now show.
   - Verify that you can import simbrief plans as per test 1.
3. Test out the FMS SimBrief import flow
   - Verify that you can import a SimBrief flight plan from the FMS
4. Test out the new "Override SimBrief User ID" feature
   - Go to Settings -> Third Party Options -> Enter a **different** user ID or SimBrief username in the field.
   - Verify it is accepted correctly.
   - Check if both EFB and FMS SimBrief import follow the **overridden ID** and not the linked Navigraph account.
   - Press the trash icon next to the field.
   - Check if both EFB and FMS SimBrief import follow the **linked Navigraph account**.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
